internal/controller/isbservicerollout_controller.go

```diff
@@ -118,25 +122,29 @@ func (r *ISBServiceRolloutReconciler) Reconcile(ctx context.Context, req ctrl.Re
 
 			statusUpdateErr := r.updateISBServiceRolloutStatusToFailed(ctx, isbServiceRollout, err)
 			if statusUpdateErr != nil {
+				r.customMetrics.ISBServiceSyncFailed.WithLabelValues().Inc()
 				return ctrl.Result{}, statusUpdateErr
 			}
-
+			r.customMetrics.ISBServiceSyncFailed.WithLabelValues().Inc()
 			return ctrl.Result{}, err
 		}
 		// restore the original status, which would've been wiped in the previous call to Update()
 		isbServiceRollout.Status = isbServiceRolloutStatus
+		r.customMetrics.ReconciliationDuration.WithLabelValues(ControllerISBSVCRollout, "update").Observe(time.Since(startTime).Seconds())
```

Comment:
does this correctly do metrics for "update" and "create"? I wouldn't think so. The decision of whether to create or update an ISBService happens within `reconcile()`, right? So, I would think you would need to pass back the information on whether there was a creation or an update from there.

Reply:
Also, what about reconciliations that result in no modification? Are we counting those or no? (I'm not saying we need to, but curious the intention)

Reply:
Unless, I'm missing something, this is not the right place for determining if we updated the child resource (which I imagine is what you're looking for, right?). This code block `if r.needsUpdate(isbServiceRolloutOrig, isbServiceRollout)` is all about whether we need to update the Rollout Spec itself. That only happens when we add a Finalizer to the Rollout because it's not already on there.

Reply:
> For "Create" I'll probably move it here
that seems good

Reply:
Thanks. Did you already sync in the latest main branch? Make sure you do since the changes you make will be very relevant to that.

Reply:
Made the changes also it's synced with main. Thanks